frozen_mutation.hh

```diff
@@ -301,7 +301,8 @@ auto frozen_mutation::consume(schema_ptr s, frozen_mutation_consumer_adaptor<Con
 
 template<FlattenedConsumerV2 Consumer>
 auto frozen_mutation::consume(schema_ptr s, Consumer& consumer, consume_in_reverse reverse) const -> frozen_mutation_consume_result<decltype(consumer.consume_end_of_stream())> {
-    frozen_mutation_consumer_adaptor adaptor(s, consumer);
+    schema_ptr ss = reverse == consume_in_reverse::yes ? s->make_reversed() : s;
```

Comment:
can we make consume_in_reverse a template parameter so the condition can be evaluated at compile time?

Reply:
I do not think it matters here. This is not a super hot code and making it a compile time parameter will increase binary size (and this is probably the single worst CPU performance issue Scylla currently has). It also doesn't quite matter, as it is very probable that the compiler already inlines the code and notices that the value of `reverse` in parent function (say, `to_data_query_result` doesn't change so it can be taken out of the loop. And also such a change would mean that I need to increase code bloat in `to_data_query_result` that would need effectively to be copy-pasted.

Reply:
Agree. If the reverse parameter ends up trickling down to add branches everywhere, then it's important, but I don't think that's the case.